### ISSUE TITLE
[question] Packages with multiple components

### ISSUE THREAD

**Issue by ytimenkov** [conan-io/conan#2288]
As I understand the solution for #14 is implemented only for boost and has tailored for CMake (new FindBoost.cmake is packaged with recipe) so impossible to use in other generators.

Are there any plans/ideas on generalizing this functionality?

I myself can imagine 2 opposite approaches:
1. Adding components into cpp_info so generators can employ it. Then CMake generator probably needs to produce a module which has support for `find_package(COMPONENTS)` and export libraries as a packages.

    However it's unclear how to introduce dependencies between components, say boost has "boost" as headers and then "boost_filesystem" just adds library and depends on "boost" to pull in headers.

2. Every generator on its own.

    In other words allow tighter coupling between package and build system it uses. Then for CMake use nice `export()` commands to produce relocatable packages, so conan's job is to define `<Package>_DIR` and `find_package(Package CONFIG ...)` will pull it in.

    Other pros of this approach:
    1. This specific configuration file may be tailored much better for the particular generator rather than generic one.
    2. For CMake in particular it greatly simplifies developing multiple components at once as developer may override where `find_package` looks for file to consume dependency directly from local build folder.

**Comment by thejohnfreeman** [comment]
See the work on components in #5090 .

**Comment by jgsogo** [comment]
I'm closing this issue, so any further conversation goes to the other issue. 💪
